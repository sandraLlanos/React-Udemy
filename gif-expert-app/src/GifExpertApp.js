import React, { useState } from 'react'
import AddCategory from './components/AddCategory';

function GifExpertApp() {
    // const categories = [ 'One Punch', 'Samurai X', 'Dragon Ball'];
    // const handleadd = () => {
        //     setCategories([ ...categories, 'HunterXHunter']);         
        //     // setCategories(cats => [ ...cats, 'HunterXHunter']);         
    // }
    const [categories, setCategories] = useState( [ 'One Punch', 'Samurai X', 'Dragon Ball'] );
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory />
            <hr/> 
            {/* <button onClick={ handleadd }>Add</button> */}
            
            <ol>
                {
                    categories.map( category => {
                        return <li key={ category }> { category } </li>;
                    })
                }   
            </ol> 

            {/* {
                categories
            }   */}
        </>
    )
}

export default GifExpertApp
