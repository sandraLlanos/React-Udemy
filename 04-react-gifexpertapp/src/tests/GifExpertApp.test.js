import React from 'react';
import { GifExpertApp } from "../GifExpertApp";
import { shallow } from "enzyme";

describe('test in <GifExpertApp /> component', () => {
    
    test('should show GifExpertApp component successfully' , () => {
        
        const wrapper = shallow( <GifExpertApp /> );
        expect(wrapper).toMatchSnapshot();
    
    });
    test('should show a list of categories', () => {
        const categories = ['One Punch', 'Goku'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories} /> );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });

});