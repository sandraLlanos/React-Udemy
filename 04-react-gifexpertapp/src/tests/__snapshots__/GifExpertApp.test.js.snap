// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <GifExpertApp /> component should show GifExpertApp component successfully 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`test in <GifExpertApp /> component should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
